### PR TITLE
docs: add generated gitignore and README for new projects

### PR DESCRIPTION
**Pull request by limaak**
Hey!

I thought it could be a little nicer if we also included a small README.md (that could grow over time) alongside a minimal .gitignore for new projects.

What do you think?

### CODE REVIEW

File: data/example/.gitignore 
Comment (Minoru) [inline data/example/.gitignore :1]: I think the default gitignore should be as minimal as possible, lest we surprise the user by hiding something they expected to see. Thus I'd only keep directories created by Hakyll, Cabal, and Stack. Also, I'd add slashes to specify that we only mean *directories* which are *in the root of the working directory*, i.e. change `_site` to `/_site/` etc. Tight and minimal.

What do you think? Is there a rationale to e.g. ignore editor-specific files like `*.swp`, platform-specific stuff like `.DS_Store`, or even generic-looking files like `tags`?

File: data/example/README.md
Comment (Minoru) [inline data/example/README.md:1]: I think this got all the key points, but I feel like they're presented in a somewhat roundabout way:

1. "Using Cabal" explains Cabal stuff in a single paragraph, then moves on to talk about working with site executable which is independent of Cabal;
2. "Using Stack" starts with an explanation of site.hs, which is independent of Stack, then continues with Stack-specific details.

To me, a better structure would be three sections:
1. "Building with Cabal"
2. "Building with Stack"
3. "Using the installed executable"

The first two sections would have a similar structure:
1. explain the `build` command
2. explain that the user either runs `cabal new-install`/`stack install` and proceeds to the third section, or they use `cabal new-run site ...`/`stack run ...` to execute commands without installing.

The third section, "Using the installed executable", would explain `site build`, `site watch`, and `site deploy`.

I'm not 100% sure this is better than what you've got. Does this proposal make sense to you?

**Review by Minoru — COMMENTED** [review]
Sorry it took me over a month to get to this! I like the overall idea, but I had to pick on some details. Please see my comments below.

File: data/example/.gitignore 
Comment (limaak) [reply data/example/.gitignore :1]: My initial idea was just copy part of hakyll [.gitignore](https://github.com/jaspervdj/hakyll/blob/master/.gitignore), but you're right we might tailor it to be more 'user' specific.

Sounds good :)

File: data/example/README.md
Comment (limaak) [reply data/example/README.md:1]: +1 for me, makes sense.

**Comment by Minoru** [comment]
Will you please make those changes, then?

**Pull request closed without merging**